--- gdb-6.1.orig/bfd/coff-alpha.c
+++ gdb-6.1/bfd/coff-alpha.c
@@ -1455,7 +1455,7 @@ alpha_relocate_section (output_bfd, info
 	  amt = sizeof (struct ecoff_section_tdata);
 	  lita_sec_data = ((struct ecoff_section_tdata *)
 			   bfd_zalloc (input_bfd, amt));
-	  ecoff_section_data (input_bfd, lita_sec) = lita_sec_data;
+	  lita_sec->used_by_bfd = lita_sec_data;
 	}
 
       if (lita_sec_data->gp != 0)
--- gdb-6.1.orig/sim/ppc/debug.c
+++ gdb-6.1/sim/ppc/debug.c
@@ -28,6 +28,7 @@
 #ifdef HAVE_STDLIB_H
 #include <stdlib.h>
 #endif
+#include <string.h>
 
 int ppc_trace[nr_trace_options];
 
